From 2e8a6ccb414ca79636392604893c79c3b0d00dc4 Mon Sep 17 00:00:00 2001
From: MarcoFalke <*~=`'#}+{/-|&$^_@721217.xyz>
Date: Sat, 4 Jul 2026 09:11:11 +0200
Subject: [PATCH] Add missing <new> includes for std::nothrow

Without the <new> include, compilation may fail:

```
src/polling_util.hpp:28:30: error: no member named 'nothrow' in namespace 'std'
   28 |             _buf = new (std::nothrow) T[nitems_];
      |                              ^~~~~~~
```
---
 src/norm_engine.cpp           | 1 +
 src/polling_util.hpp          | 1 +
 src/proxy.cpp                 | 1 +
 src/reaper.cpp                | 1 +
 src/session_base.cpp          | 1 +
 src/socket_poller.cpp         | 1 +
 src/stream_connecter_base.cpp | 1 +
 src/stream_listener_base.cpp  | 2 ++
 src/ws_engine.cpp             | 1 +
 9 files changed, 10 insertions(+)

diff --git a/src/norm_engine.cpp b/src/norm_engine.cpp
index 8bb3da7d95..f299b63058 100644
--- a/src/norm_engine.cpp
+++ b/src/norm_engine.cpp
@@ -1,6 +1,7 @@
 
 #include "precompiled.hpp"
 
+#include <new>
 #include "platform.hpp"
 
 #if defined ZMQ_HAVE_NORM
diff --git a/src/polling_util.hpp b/src/polling_util.hpp
index 13a4911ffb..3536be9a46 100644
--- a/src/polling_util.hpp
+++ b/src/polling_util.hpp
@@ -4,6 +4,7 @@
 #define __ZMQ_SOCKET_POLLING_UTIL_HPP_INCLUDED__
 
 #include <stdlib.h>
+#include <new>
 #include <vector>
 
 #if defined ZMQ_HAVE_WINDOWS
diff --git a/src/proxy.cpp b/src/proxy.cpp
index 5ba77b3775..8cbfc8c8a1 100644
--- a/src/proxy.cpp
+++ b/src/proxy.cpp
@@ -3,6 +3,7 @@
 #include "precompiled.hpp"
 
 #include <stddef.h>
+#include <new>
 #include "poller.hpp"
 #include "proxy.hpp"
 #include "likely.hpp"
diff --git a/src/reaper.cpp b/src/reaper.cpp
index 4361a7e97b..1484a926cb 100644
--- a/src/reaper.cpp
+++ b/src/reaper.cpp
@@ -2,6 +2,7 @@
 
 #include "precompiled.hpp"
 #include "macros.hpp"
+#include <new>
 #include "reaper.hpp"
 #include "socket_base.hpp"
 #include "err.hpp"
diff --git a/src/session_base.cpp b/src/session_base.cpp
index bfc25d165f..e47745ecea 100644
--- a/src/session_base.cpp
+++ b/src/session_base.cpp
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: MPL-2.0 */
 
 #include "precompiled.hpp"
+#include <new>
 #include "macros.hpp"
 #include "session_base.hpp"
 #include "i_engine.hpp"
diff --git a/src/socket_poller.cpp b/src/socket_poller.cpp
index b9eab821a8..30bde2ff43 100644
--- a/src/socket_poller.cpp
+++ b/src/socket_poller.cpp
@@ -7,6 +7,7 @@
 #include "macros.hpp"
 
 #include <limits.h>
+#include <new>
 
 static bool is_thread_safe (const zmq::socket_base_t &socket_)
 {
diff --git a/src/stream_connecter_base.cpp b/src/stream_connecter_base.cpp
index 8dd9a6d69f..6b733b4d92 100644
--- a/src/stream_connecter_base.cpp
+++ b/src/stream_connecter_base.cpp
@@ -15,6 +15,7 @@
 #endif
 
 #include <limits>
+#include <new>
 
 zmq::stream_connecter_base_t::stream_connecter_base_t (
   zmq::io_thread_t *io_thread_,
diff --git a/src/stream_listener_base.cpp b/src/stream_listener_base.cpp
index 350093ad01..b4e02027de 100644
--- a/src/stream_listener_base.cpp
+++ b/src/stream_listener_base.cpp
@@ -13,6 +13,8 @@
 #include <winsock2.h>
 #endif
 
+#include <new>
+
 zmq::stream_listener_base_t::stream_listener_base_t (
   zmq::io_thread_t *io_thread_,
   zmq::socket_base_t *socket_,
diff --git a/src/ws_engine.cpp b/src/ws_engine.cpp
index 9eec0d929c..55590cdca5 100644
--- a/src/ws_engine.cpp
+++ b/src/ws_engine.cpp
@@ -26,6 +26,7 @@
 #endif
 
 #include <cstring>
+#include <new>
 
 #include "compat.hpp"
 #include "tcp.hpp"
